ruma-api-macros/src/api/metadata.rs

```diff
@@ -61,6 +61,13 @@ impl TryFrom<RawMetadata> for Metadata {
                 },
                 "path" => match expr {
                     Expr::Lit(ExprLit { lit: Lit::Str(literal), .. }) => {
+                        let path_str = literal.value();
+                        if !util::is_ascii_printable(&path_str) || path_str.contains(' ') {
```

Comment:
We could just update `is_ascii_printable` to check `0x21` onwards instead of `0x20` and rename it to sth. like `is_valid_endpoint_path`.

Reply:
Yeah I like that I just couldn't think of a good name to call it but `is_valid_endpoint_path` that is perfect.